### DISCUSSION TITLE
Inertia-React v18 Laravel compatibility problem

### DISCUSSION THREAD

**Discussion by dvelez3815** [inertiajs/inertia#1167] [Help (React)]
Hi, I am currently working with inertiajs/inertia-laravel": "^0.5.4" and using the new versions of react and react-dom:18 the problem I am facing is by following the documentation 

![Documentation](https://user-images.githubusercontent.com/40648189/166490603-032e3ee0-c46a-4157-bf20-9431ec37aef0.png)

I get the next warning. So render it not working properly

![problem_](https://user-images.githubusercontent.com/40648189/166491146-31d65024-8e16-4c0b-8ca3-16014eafd206.png)

I was trying to use react-dom/client instead of react-dom but again more problem. Has anyone faced this problem recently and solved it? It it's possible to upgrade the web documentation based on new react version? Thanks in advanced

**Comment by sideshowlol** [comment]
I got it to work like this:

```
import React from 'react'
import { createRoot } from "react-dom/client"
import { createInertiaApp } from '@inertiajs/inertia-react'

createInertiaApp({
  resolve: name => import(`./Pages/${name}`),
  setup({ el, App, props }) {
    const root = createRoot(el);
    root.render(<App {...props} />)
  },
})
```

Hope that helps.
